app/javascript/packs/feedPreviewCards.jsx

```diff
@@ -0,0 +1,109 @@
+import { h, render } from 'preact';
+import { UserMetadata } from '../profilePreviewCards/UserMetadata';
+import { initializeDropdown } from '@utilities/dropdownUtils';
+
+async function populateMissingMetadata(metadataPlaceholder) {
+  const { authorId } = metadataPlaceholder.dataset;
+  const response = await fetch(`/profile_preview_cards/${authorId}`, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    credentials: 'same-origin',
+  });
+  const authorMetadata = await response.json();
+
+  // A given author may have multiple cards in the feed - populate their metadata everywhere it is missing
+  const allPlaceholdersForThisAuthor = document.querySelectorAll(
+    `.author-preview-metadata-container[data-author-id="${authorId}"]`,
+  );
+
+  for (const placeholder of allPlaceholdersForThisAuthor) {
+    render(
```

Comment:
Refactor (potentially blocking): Can we render on mouseover instead? That way we only add it as we need it. I'm not positive if this will cause accessibility issues though. Regardless of that, this is a perfect use case for [memo](https://reactjs.org/docs/react-api.html#reactmemo) on the component.